src/platform_impl/windows/util.rs

```diff
@@ -205,6 +208,21 @@ pub fn is_focused(window: HWND) -> bool {
     window == unsafe { GetActiveWindow() }
 }
 
+pub fn get_instance_handle() -> HINSTANCE {
+    // Gets the instance handle by taking the address of the
+    // pseudo-variable created by the microsoft linker:
+    // https://devblogs.microsoft.com/oldnewthing/20041025-00/?p=37483
+
+    // This is preferred over GetModuleHandle(NULL) because it also works in DLLs:
+    // https://stackoverflow.com/questions/21718027/getmodulehandlenull-vs-hinstance
+
+    extern "C" {
+        static __ImageBase: IMAGE_DOS_HEADER;
+    }
+
+    unsafe { &__ImageBase as *const IMAGE_DOS_HEADER as _ }
```

Comment:
That gives me the following error:

```
211 | pub fn get_instance_handle() -> HINSTANCE {
    |                                 --------- expected `isize` because of return type
...
223 |     unsafe { <*const _>::cast(&__ImageBase) }
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `isize`, found *-ptr```

`windows_sys` redefines `HINSTANCE` to be isize, so I guess you'd need to cast a reference as isize if that works somehow, good idea though!

Reply:
Ah, right, it's an `isize` and not a pointer. My bad.

Of the options you presented, I think the last one is the "best".

Reply:
Fair enough, I pushed a commit to change it.